### PR TITLE
fix: add more linked panels for attributes

### PR DESCRIPTION
**Pull request by teolemon**
### What

- adds more linked knowledge panels for various attributes
  - Allergens (commented out, since we don't have an associated knowledge panel yet)
  - vegan/vegetarian/palm oil unknown

### Sample query
  - https://world.openfoodfacts.org/api/v2/product/4006544205006.json?fields=knowledge_panels,attributes

### Why
- This is important because it increases the likelihood that someone will take a photo of/extract the ingredient list

### Related issue(s) and discussion
- #5515 

https://user-images.githubusercontent.com/1689815/225931746-87b11687-1d27-456a-93a2-b3384e9dea51.mp4

### CODE REVIEW

File: lib/ProductOpener/Attributes.pm [hunk 1]
```diff
@@ -336,7 +340,7 @@ sub initialize_attribute ($attribute_id, $target_lc) {
 			my $name = display_taxonomy_tag($target_lc, "ingredients_analysis", "en:$analysis_tag");
 			$attribute_ref->{name} = $name;
 			$attribute_ref->{setting_name} = $name;
-
+			$attribute_ref->{panel_id} = "ingredients_analysis_en-" . $analysis_tag;
```
Comment (alexgarel) [inline lib/ProductOpener/Attributes.pm:343]: I think it's:
```suggestion
		$attribute_ref->{panel_id} = "ingredients_analysis_en:" . $analysis_tag;
```
eg `ingredients_analysis_en:palm-oil-free`

File: lib/ProductOpener/Attributes.pm [hunk 2]
```diff
@@ -299,13 +301,15 @@ sub initialize_attribute ($attribute_id, $target_lc) {
...
+		$attribute_ref->{panel_id} = "ingredients_analysis_en-" . $analysis_tag;
+		# does not seem to be working
 	}
 	elsif ($attribute_id =~ /^(labels)_(.*)$/) {
 		my $tagtype = $1;
 		my $tag = $2;
 		$tag =~ s/_/-/g;
-
 		$attribute_ref->{icon_url} = "$static_subdomain/images/attributes/${tag}.svg";
+		$attribute_ref->{panel_id} = "${tag}";
```
Comment (alexgarel) [inline lib/ProductOpener/Attributes.pm:312]: There is "environment_label_en:" in front. And we should do this only for tags with a panel (how do we get them ? Is it through a property in taxonomy ?)

**Review by alexgarel — CHANGES_REQUESTED** [review]
Cool idea, but it needs a bit of work

File: lib/ProductOpener/Attributes.pm (hunk 2 above)
Comment (teolemon) [reply lib/ProductOpener/Attributes.pm:311]: @alexgarel would removing this change for the labels make the PR acceptable to you ? we indeed don't have any knowledge panels on labels, but that would be another issue that we can create

File: lib/ProductOpener/Attributes.pm [hunk 3]
```diff
@@ -299,13 +301,14 @@ sub initialize_attribute ($attribute_id, $target_lc) {
...
 		$attribute_ref->{icon_url} = "$static_subdomain/images/attributes/$analysis_tag.svg";
+		$attribute_ref->{panel_id} = "ingredients_analysis_en:" . $analysis_tag;
 	}
 	elsif ($attribute_id =~ /^(labels)_(.*)$/) {
 		my $tagtype = $1;
 		my $tag = $2;
 		$tag =~ s/_/-/g;
-
 		$attribute_ref->{icon_url} = "$static_subdomain/images/attributes/${tag}.svg";
+		$attribute_ref->{panel_id} = "${tag}";
```
Comment (teolemon) [inline lib/ProductOpener/Attributes.pm:311]:
```suggestion
        # Will be relevant when we have panels for all labels
		# $attribute_ref->{panel_id} = "${tag}";
```

**Comment by teolemon** [comment]
@stephanegigandet awesome, I re-requested a review from @alexgarel or you can self approve your changes 👍 
poke @g123k

**Comment by teolemon** [comment]
image

**Review by stephanegigandet — APPROVED** [review]

**Comment by teolemon** [comment]
@alexgarel your changes should be addressed, can you approve ?

**Review by alexgarel — APPROVED** [review]

**Pull request merged**